pkg/service/main.go

```diff
@@ -98,37 +99,41 @@ type TriggerResult struct {
 // However, we keep it here for now because we may need it in the future.
 // service is the implementation of the Service interface
 type service struct {
-	repository               repository.Repository
-	redisClient              *redis.Client
-	temporalClient           client.Client
-	component                *componentstore.Store
-	mgmtPublicServiceClient  mgmtpb.MgmtPublicServiceClient
-	mgmtPrivateServiceClient mgmtpb.MgmtPrivateServiceClient
-	aclClient                acl.ACLClientInterface
-	converter                Converter
-	minioClient              miniox.MinioI
-	memory                   memory.MemoryStore
-	log                      *zap.Logger
-	workerUID                uuid.UUID
-	retentionHandler         MetadataRetentionHandler
-	binaryFetcher            external.BinaryFetcher
+	repository                   repository.Repository
+	redisClient                  *redis.Client
+	temporalClient               client.Client
+	component                    *componentstore.Store
+	mgmtPublicServiceClient      mgmtpb.MgmtPublicServiceClient
+	mgmtPrivateServiceClient     mgmtpb.MgmtPrivateServiceClient
+	aclClient                    acl.ACLClientInterface
+	converter                    Converter
+	minioClient                  miniox.MinioI
+	memory                       memory.MemoryStore
+	log                          *zap.Logger
+	workerUID                    uuid.UUID
+	retentionHandler             MetadataRetentionHandler
+	binaryFetcher                external.BinaryFetcher
+	artifactPublicServiceClient  artifactpb.ArtifactPublicServiceClient
+	artifactPrivateServiceClient artifactpb.ArtifactPrivateServiceClient
```

Comment:
❓ Is this one used?

Reply:
Yes, it is used in blobstorage. In the future, I think it will be better to move it to repository layer if possible.